plugins/modules/ndo_l3out_routed_interface.py

```diff
@@ -0,0 +1,544 @@
+          template_id:
+            description:
+            - The ID of the template that contains the port channel.
+            - This parameter or O(port_channel.reference.template) is required.
+            type: str
+      micro_bfd_enabled:
+        description:
+        - Whether to enable micro BFD (Bidirectional Forwarding Detection) on the interface.
+        type: bool
+      micro_bfd_address:
+        description:
+        - The address for micro BFD.
+        type: str
+      micro_bfd_start_timer:
+        description:
+        - The start timer for micro BFD.
+        - The value must be 0 or in the range 60 - 3600.
+        type: int
+  interface_group_policy:
+    description:
+    - The name of the interface group policy.
+    type: str
+  ipv4_address:
+    description:
+    - The IPv4 address of the interface.
+    type: str
+  ipv6_address:
+    description:
+    - The IPv6 address of the interface.
+    type: str
+  ipv6_link_local_address:
+    description:
+    - The IPv6 link-local address of the interface.
+    type: str
+  ipv6_dad:
+    description:
+    - Whether to enable IPv6 Duplicate Address Detection (DAD).
+    - If this parameter is unspecified, NDO defaults to O(ipv6_dad=enabled).
+    type: str
+    choices: [ enabled, disabled ]
+  mac:
+    description:
+    - The MAC address of the interface.
+    type: str
+  mtu:
+    description:
+    - The Maximum Transmission Unit (MTU) of the interface.
+    - Use O(mtu=inherit) to inherit the value configured under the fabric L2 MTU settings.
+    - The value must be 1, in the range 576 - 9216 or O(mtu=inherit).
+    type: str
+  target_dscp:
+    description:
+    - The target Differentiated Services Code Point (DSCP) of the interface.
+    - If this parameter is unspecified, NDO defaults to O(target_dscp=unspecified).
+    type: str
+    choices:
+    - af11
+    - af12
+    - af13
+    - af21
+    - af22
+    - af23
+    - af31
+    - af32
+    - af33
+    - af41
+    - af42
+    - af43
+    - cs0
+    - cs1
+    - cs2
+    - cs3
+    - cs4
+    - cs5
+    - cs6
+    - cs7
+    - expedited_forwarding
+    - unspecified
+    - voice_admit
+  state:
+    description:
+    - Determines the desired state of the resource.
+    - Use C(absent) to remove the resource.
+    - Use C(query) to list the resource.
+    - Use C(present) to create or update the resource.
+    type: str
+    choices: [ absent, query, present ]
+    default: query
+notes:
+- The O(template) or O(template_id) must exist before using this module in your playbook.
+  The M(cisco.mso.ndo_template) module can be used for this.
+- The O(l3out) or O(l3out_uuid) must exist before using this module in your playbook.
+  The M(cisco.mso.ndo_l3out_template) module can be used for this.
+seealso:
+- module: cisco.mso.ndo_template
+- module: cisco.mso.ndo_l3out_template
+extends_documentation_fragment: cisco.mso.modules
+"""
+
+EXAMPLES = r"""
+- name: Create a L3Out Routed Interface of type port
```

Comment:
```suggestion
- name: Create a L3Out Routed Interface of type port and a L3Out Node attached to it
```
something like that to be clearer

Reply:
Let me know if this is more clear for you. I chose other wording because the L3Out node is not attached to the routed interface. It is configuration that needs to be in place in order for a interface to be configured.

Reply:
your wording is better, that's definitely clearer, thanks